airbyte-cdk/python/airbyte_cdk/destinations/destination.py

```diff
@@ -104,18 +106,24 @@ def parse_args(self, args: List[str]) -> argparse.Namespace:
 
     def run_cmd(self, parsed_args: argparse.Namespace) -> Iterable[AirbyteMessage]:
         cmd = parsed_args.command
+        if cmd not in self.VALID_CMDS:
+            raise Exception(f"Unrecognized command: {cmd}")
+
+        spec = self.spec(self.logger)
         if cmd == "spec":
-            yield self._run_spec()
-        elif cmd == "check":
-            yield self._run_check(config_path=parsed_args.config)
+            yield self._run_spec(spec)
+            return
+        config = self.read_config(config_path=parsed_args.config)
+        check_error_msg = check_config_against_spec(config, spec)
+        if check_error_msg:
```

Comment:
an invalid config should cause the exit code to be >0, it's invalid input. An exit code of 0 means the connector succeeded

Reply:
Updated